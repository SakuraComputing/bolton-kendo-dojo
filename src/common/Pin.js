import React from 'react';

const Pin = () => {
    return (
        <div>
            <div className="marker">
                <div className="pin"></div>
                <div className="pin-effect"></div>
            </div>             
        </div>
    );
}
export default Pin;