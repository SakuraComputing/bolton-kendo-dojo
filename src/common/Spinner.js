import React from 'react';
import spinner from '../../public/img/util/spinner.gif';

const Spinner = () => {
  return (
    <div>
      <img 
        className="spinner"
        src={spinner}
        alt="Loading..."
      />
    </div>
  )
}
export default Spinner;
