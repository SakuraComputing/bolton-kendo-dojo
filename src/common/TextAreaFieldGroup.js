import React from 'react';
import PropTypes from 'prop-types';

export const TextAreaFieldGroup = ({
    name,
    placeholder,
    value,
    error,
    info,
    onChange,

}) => {
    return (
        <div className="form-group">
            <textarea
                className="form-control-large"
                placeholder={placeholder}
                name={name}
                value={value}
                onChange={onChange}
                rows="3"
            />
            {info && <small className="form-text text-muted">{info}</small>}
            {error && (<div className="invalid-feedback">{error}</div>)}
        </div>
    )
};

TextAreaFieldGroup.propTypes = {
    name: PropTypes.string.isRequired,
    placeholder: PropTypes.string,
    value: PropTypes.string.isRequired,
    error: PropTypes.string,
    info: PropTypes.string,
    onChange: PropTypes.func.isRequired,
};


export default TextAreaFieldGroup;
