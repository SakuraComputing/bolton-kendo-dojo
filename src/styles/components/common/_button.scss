.button {
    border: solid 1px $blue-black;
    padding: .9em;
    border-radius: 5px;
    font-size: 100%;
    font-weight: bold;
    width: 130px;
}

.button-info,
.button-small {
    margin-top: 1em;
    color: $blue-black;
    background-color: $light-orange;
    transition: 0.5s;
    outline-color: $dark-orange;
}

.button-info:hover,
.button-small:hover {
    background-color: #ffc91a;
}

.button-small {
    border: solid 1px $blue-black;
    padding: .5em;
    border-radius: 5px;
    font-size: 80%;
    font-weight: bold;
    width: 100px;
    cursor: pointer;
}

.button-delete {
    background-color: red;
    color: $off-white;
}