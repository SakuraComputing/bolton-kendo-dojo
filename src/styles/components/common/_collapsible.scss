.wrap-collabsible {
    margin-bottom: 1.2rem 0;
  }
  
  input[type='checkbox'] {
    display: none;
  }
  
  .lbl-toggle-search {
    display: block;
    font-weight: bold;
    font-size: 100%;
    text-align: center;
    padding: 1rem;
    color: $dark-orange;
    background: $light-orange;
    cursor: pointer;
    border-radius: 7px;
    transition: all 0.25s ease-out;
  }
  
  .lbl-toggle-upload {
    display: block;
    font-weight: bold;
    font-size: 100%;
    text-align: center;
    padding: 1rem;
    color: $dark-orange;
    background: $light-orange;
    cursor: pointer;
    border-radius: 7px;
    transition: all 0.25s ease-out;
  }
  
  .lbl-toggle-search:hover {
    color: $dark-orange;
  }
  
  .lbl-toggle-upload:hover {
    color: $dark-orange;
  }
  
  .lbl-toggle-search::before {
    content: ' ';
    display: inline-block;
  
    border-top: 5px solid transparent;
    border-bottom: 5px solid transparent;
    border-left: 5px solid currentColor;
    vertical-align: middle;
    margin-right: .7rem;
    transform: translateY(-2px);
  
    transition: transform .2s ease-out;
  }

  .lbl-toggle-upload::before {
    content: ' ';
    display: inline-block;
  
    border-top: 5px solid transparent;
    border-bottom: 5px solid transparent;
    border-left: 5px solid currentColor;
    vertical-align: middle;
    margin-right: .7rem;
    transform: translateY(-2px);
  
    transition: transform .2s ease-out;
  }
  
  .toggle-search:checked + .lbl-toggle-search::before {
    transform: rotate(90deg) translateX(-3px);
  }
  
  .toggle-upload:checked + .lbl-toggle-upload::before {
    transform: rotate(90deg) translateX(-3px);
  }
  
  .collapsible-content-search {
    max-height: 0px;
    overflow: hidden;
    transition: max-height .25s ease-in-out;
  }
  
  .collapsible-content-upload {
    max-height: 0px;
    overflow: hidden;
    transition: max-height .25s ease-in-out;
  }
  
  .toggle-search:checked + .lbl-toggle-search + .collapsible-content-search {
    max-height: 350px;
  }
  
  .toggle-upload:checked + .lbl-toggle-upload + .collapsible-content-upload {
    max-height: 350px;
  }
  
  .toggle-search:checked + .lbl-toggle-search {
    border-bottom-right-radius: 0;
    border-bottom-left-radius: 0;
  }
  
  .toggle-upload:checked + .lbl-toggle-upload {
    border-bottom-right-radius: 0;
    border-bottom-left-radius: 0;
  }
  
  .collapsible-content .content-inner {
    background: $off-white;
    border-bottom: $dark-orange;
    border-bottom-left-radius: 7px;
    border-bottom-right-radius: 7px;
    padding: .5rem 1rem;
  }