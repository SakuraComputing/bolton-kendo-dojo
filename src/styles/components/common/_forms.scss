.form-group {
    padding: 10px;
}

.form-title {
    font-size: $xl-size;
    font-weight: 200;
    padding-top: 1em;
}

.form-text {
    font-size: 1.8em;
}

.form-control-large {
    width: 100%;
    font-size: 110%;
}

.input-box {
    padding: 8px;
    border: solid 1px $dark-grey;
    border-radius: 5px;
    font-size: 100%;
    transition: 0.3s;
    outline: none;
}

.input-box:hover {
  background-color: $light-blue;
}

// error messsgae
.invalid-feedback {
    color: $warning-red;
}

.spinner-container {
    background-color: $off-white;
    height: 75%;
    width: 80%;
    z-index: 999;
    overflow: hidden;
    position: absolute;
}

.spinner {
    position: absolute;
    top:0;
    bottom: 0;
    left: 0;
    right: 0;
    margin: auto;
}


