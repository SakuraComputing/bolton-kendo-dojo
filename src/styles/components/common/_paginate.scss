.pagination {
    width: 100%;

    ul {
        display: flex;
        flex-direction: row;
        flex-wrap: wrap;
    }

    li {
        list-style: none;
        display: inline-block;
        padding: 10px 20px;
        justify-content: center;
        align-items: center;
        border: solid 1px $blue_black;
        color: $blue-black;
        background-color: $light-orange;
        transition: 0.2s;
    }

    li:active {
        background-color: $color__primary;
    }

    li:hover {
        background-color: $dark-orange;
    }

    .previous_page, .next_page {
        outline: none;
    }
}

.active {
    background-color: $color__primary;
}



  