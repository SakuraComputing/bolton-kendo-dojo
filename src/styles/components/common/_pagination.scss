.pagination {

    padding-left: 0;
    width: 100%;
    height: 40px;

    ul {
        display: flex;
        flex-direction: row;
        flex-wrap: wrap;
    }

    li {
        list-style: none;
        display: inline-block;
        justify-content: center;
        align-items: center;
        border-right: solid 1px $blue_black;
        border-top: solid 1px $blue_black;
        border-bottom: solid 1px $blue_black;

        color: $blue-black;
        background-color: $off-white;
        transition: 0.2s;
        height: 100%;
        
        a { 
            display: inline-block;
            padding: 7px 20px;
            outline: none;
        }
    }

    li:first-child {
        border-left: solid 1px $blue-black;
        border-top-left-radius: 3px;
        border-bottom-left-radius: 3px;
    }

    li:last-child {
        border-top-right-radius: 3px;
        border-bottom-right-radius: 3px;
    }

    li:active {
        background-color: $color__primary;
        color: $off-white;
    }

    .previous_page, .next_page {
        outline: none;
    }
    .active {
        background-color: $color__primary;
        color: $off-white;
    }    
    .disabled {
        color: $grey;
    }

    .disable:hover {
        background-color: $off-white !important;
    }
}



  