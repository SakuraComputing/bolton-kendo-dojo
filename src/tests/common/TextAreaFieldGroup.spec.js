import React from 'react';
import { shallow } from 'enzyme';
import { TextAreaFieldGroup } from '../../common/TextAreaFieldGroup';

describe('Text Area Field Group', () => {

    let wrapper, name, value, type, onChange;

    beforeEach(() => {
        name = 'test';
        value = 'test';
        type = 'text';
        onChange = jest.fn();

        wrapper = shallow(<TextAreaFieldGroup 
            name={name} 
            value={value}
            type={type}
            onChange={onChange}
        />);
    })
    it('should render correctly', () => {
        expect(wrapper).toMatchSnapshot();
    });
});