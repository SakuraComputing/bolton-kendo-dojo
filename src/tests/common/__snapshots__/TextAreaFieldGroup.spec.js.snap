// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Text Area Field Group should render correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <TextAreaFieldGroup
    name="test"
    onChange={[MockFunction]}
    type="text"
    value="test"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <textarea
          className="form-control-large"
          name="test"
          onChange={[MockFunction]}
          rows="3"
          value="test"
        />,
        undefined,
        undefined,
      ],
      "className": "form-group",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "className": "form-control-large",
          "name": "test",
          "onChange": [MockFunction],
          "placeholder": undefined,
          "rows": "3",
          "value": "test",
        },
        "ref": null,
        "rendered": null,
        "type": "textarea",
      },
      undefined,
      undefined,
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <textarea
            className="form-control-large"
            name="test"
            onChange={[MockFunction]}
            rows="3"
            value="test"
          />,
          undefined,
          undefined,
        ],
        "className": "form-group",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "form-control-large",
            "name": "test",
            "onChange": [MockFunction],
            "placeholder": undefined,
            "rows": "3",
            "value": "test",
          },
          "ref": null,
          "rendered": null,
          "type": "textarea",
        },
        undefined,
        undefined,
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
